cli/lib/site-utils.ts

```diff
@@ -83,20 +83,31 @@ export async function setupCustomDomain(
  * @param stoppedSiteId - The ID of the site that was just stopped (to exclude from the check)
  */
 export async function stopProxyIfNoSitesNeedIt(
-	stoppedSiteId: string,
+	stoppedSiteIds: string | string[],
 	logger: Logger< LoggerAction >
 ): Promise< void > {
+	const stoppedSiteIdsArray = Array.isArray( stoppedSiteIds ) ? stoppedSiteIds : [ stoppedSiteIds ];
+
 	const proxyProcess = await isProxyProcessRunning();
 	if ( ! proxyProcess ) {
 		return;
 	}
 
 	const appdata = await readAppdata();
 
-	for ( const site of appdata.sites ) {
-		if ( site.id !== stoppedSiteId && site.customDomain && ( await isServerRunning( site.id ) ) ) {
-			return;
-		}
+	const remainingSites = appdata.sites.filter(
```

Comment:
It takes some time to read the condition, especially with adding `[].includes`, so I plit it to separate conditions with variables